hack/generator/pkg/astmodel/package_definition.go

```diff
@@ -33,6 +34,10 @@ func (pkgDef *PackageDefinition) AddDefinition(def TypeDefiner) {
 	pkgDef.definitions = append(pkgDef.definitions, def)
 }
 
+// TODO: I feel like we should move a lot of this "create the files" stuff out of package_definition because
```

Comment:
Thoughts on this?

Reply:
one thing is we would have to make the types have more robust public interfaces since we can't define 'member functions' in separate packages

Reply:
I've removed this comment (for cleanliness), and will poke around with it locally.